### PR TITLE
Addressing issues #34, #35 and https://github.com/cms-gem-daq-project/xhal/pull/58 

### PR DESCRIPTION
**Pull request by mexanick**
Addressing issues #34, #35 and adding the LINKX_UNDERFLOW counter to TRIGGEROHMain

## Description

Fixing the `switch` statements and adding missing monitoring registers. Documentation is updated thanks to @evka85 
### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by bdorney** [comment]
I'd like this to be merged after #37 to prevent repeat testing (again).  Additionally there are several PR's in the other repo's that #37 depends on.

File: src/calibration_routines.cpp [hunk 1]
```diff
@@ -659,87 +681,98 @@ void genScan(const RPCMsg *request, RPCMsg *response)
...
+            uint32_t vfatN = (invertVFATPos) ? 23 - (*vfatNptr).second : (*vfatNptr).second;
 
-    uint32_t goodVFATs = vfatSyncCheckLocal(la, ohN);
-    if( !( (goodVFATs >> vfatN ) & 0x1 ) ){
-        sprintf(regBuf,"The requested VFAT is not synced; goodVFATs: %x\t requested VFAT: %i; maskOh: %x", goodVFATs, vfatN, maskOh);
-        la->response->set_string("error",regBuf);
-        return;
-    }
+            uint32_t goodVFATs = vfatSyncCheckLocal(la, ohN);
+            if( !( (goodVFATs >> vfatN ) & 0x1 ) ){
```
Comment (bdorney) [inline src/calibration_routines.cpp:724]: I think your editor is inserting `\t` which can cause formatting issues.  Is it possible you update your `~/.vimrc` file or similar to replace `\t` with four `\s` characters?

**Review by bdorney — CHANGES_REQUESTED** [review]
Aesthetic change requested to ensure no problems with different editors are introduced.

File: src/calibration_routines.cpp (hunk 1 above)
Comment (mexanick) [reply src/calibration_routines.cpp:724]: @bdorney I believe the new shift here is correct as I have changed a an `if then else` statement to a `switch` statement in this function which requires another indentation shift. See line 684. I have checked the file and it appears to have all the tabs converted to spaces, e.g. there's no \t but four \s everywhere

**Review by bdorney — APPROVED** [review]

**Pull request merged**